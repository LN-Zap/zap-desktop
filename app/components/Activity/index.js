import Activity from './Activity'

export default Activity
