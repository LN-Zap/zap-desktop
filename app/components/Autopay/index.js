import Autopay from './Autopay'

export default Autopay
export AutopayActions from './AutopayActions'
export AutopayCardView from './AutopayCardView'
export AutopayHeader from './AutopayHeader'
export AutopayHeading from './AutopayHeading'
export AutopayList from './AutopayList'
export AutopaySearch from './AutopaySearch'
export AutopaySearchNoResults from './AutopaySearchNoResults'
