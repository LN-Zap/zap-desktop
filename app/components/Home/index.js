export Home from './Home'
export CreateWalletButton from './CreateWalletButton'
export NoWallets from './NoWallets'
export WalletsMenu from './WalletsMenu'
export WalletLauncher from './WalletLauncher'
export WalletUnlocker from './WalletUnlocker'
export WalletSettingsFormLocal from './WalletSettingsFormLocal'
export WalletSettingsFormRemote from './WalletSettingsFormRemote'
export WalletHeader from './WalletHeader'
