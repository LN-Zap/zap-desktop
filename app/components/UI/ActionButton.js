import React, { useState, useEffect, useRef } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import delay from '@zap/utils/delay'
import Button from './Button'
import Spinner from './Spinner'

const StyledButton = styled(Button)`
  color: ${props => (props.active ? props.theme.colors.lightningOrange : null)};
  &:hover {
    color: ${props => props.theme.colors.lightningOrange};
  }
`
StyledButton.propTypes = {
  active: PropTypes.bool,
}

const ActionButton = ({ children, hint, onClick, timeout, ...rest }) => {
  const [status, setStatus] = useState(null)
  const buttonRef = useRef()

  async function triggerAction() {
    await onClick()
    await delay(timeout)
    setStatus('done')
  }

  useEffect(() => {
    if (status === 'fetching') {
      triggerAction()
    } else if (status === 'done') {
      buttonRef.current.blur()
    }
  }, [status])

  function handleClick() {
    setStatus('fetching')
  }

  return (
    <StyledButton
      ref={buttonRef}
      active={status === 'fetching'}
      className="hint--bottom-left"
      data-hint={hint}
      onClick={handleClick}
      size="small"
      variant="secondary"
      {...rest}
    >
      {status === 'fetching' ? <Spinner /> : children}
    </StyledButton>
  )
}

ActionButton.propTypes = {
  children: PropTypes.node,
  hint: PropTypes.node,
  onClick: PropTypes.func.isRequired,
  timeout: PropTypes.number,
}

ActionButton.defaultProps = {
  timeout: 1000,
}

export default ActionButton
