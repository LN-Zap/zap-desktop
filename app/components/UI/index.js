export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundTertiary from './BackgroundTertiary'
export BackgroundSecondary from './BackgroundSecondary'
export Bar from './Bar'
export Button from './Button'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export DataRow from './DataRow'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export Message from './Message'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export MainContent from './MainContent'
export Menu from './Menu'
export MenuItem from './MenuItem'
export MenuItemGroup from './MenuItemGroup'
export Modal from './Modal'
export Notification from './Notification'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export Page from './Page'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Truncate from './Truncate'
export Value from './Value'
export Wizard from './Wizard'
