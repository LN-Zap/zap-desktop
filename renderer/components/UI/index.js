export ActionBar from './ActionBar'
export ActionButton from './ActionButton'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export { BasicInput } from './Input'
export { BasicRange } from './Range'
export { BasicTextArea } from './TextArea'
export { BasicCheckbox } from './Checkbox'
export { BasicSelect } from './Select'
export Button from './Button'
export Card from './Card'
export Checkbox from './Checkbox'
export CloseButton from './CloseButton'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export DialogOverlay from './DialogOverlay'
export Donut from './Donut'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export LndConnectionStringInput from './LndConnectionStringInput'
export LndConnectionStringEditor from './LndConnectionStringEditor'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export { ModalOverlayStyles } from './Modal'
export NodePubkeyInput from './NodePubkeyInput'
export Notification from './Notification'
export OpenDialogButton from './OpenDialogButton'
export OpenDialogInput from './OpenDialogInput'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export ProgressBar from './ProgressBar'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioCard from './RadioCard'
export RadioGroup from './RadioGroup'
export Range from './Range'
export RowLabel from './RowLabel'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Tab from './Tab'
export Tabs from './Tabs'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Tooltip from './Tooltip'
export TransactionFeeInput from './TransactionFeeInput'
export Value from './Value'
export Wizard from './Wizard'
