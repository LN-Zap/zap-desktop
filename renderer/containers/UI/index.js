export CurrencyFieldGroup from './CurrencyFieldGroup'
export CryptoSelector from './CryptoSelector'
export CryptoValue from './CryptoValue'
export FiatSelector from './FiatSelector'
export FiatValue from './FiatValue'
