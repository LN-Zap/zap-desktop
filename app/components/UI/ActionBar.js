import React from 'react'
import PropTypes from 'prop-types'
import { Flex } from 'rebass'
import { Button } from 'components/UI'

const createButtons = buttons =>
  buttons.map((entry, index) => (
    <Button
      key={entry.name}
      variant={index === buttons.length - 1 ? 'normal' : 'secondary'}
      mr={index < buttons.length - 1 ? 6 : 0}
      onClick={entry.onClick}
    >
      {entry.name}
    </Button>
  ))

const ActionBar = ({ isOpen, buttons, ...rest }) => {
  // check if buttons is a descriptive array  or a React renderable node
  const buttonsLayout = React.isValidElement(buttons) ? buttons : createButtons(buttons)

  return (
    <Flex
      flexDirection="row"
      justifyContent="flex-end"
      bg="secondaryColor"
      pr={5}
      pt={2}
      pb={2}
      {...rest}
    >
      {buttonsLayout}
    </Flex>
  )
}

ActionBar.propTypes = {
  buttons: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired
      })
    )
  ]).isRequired
}

export default ActionBar
