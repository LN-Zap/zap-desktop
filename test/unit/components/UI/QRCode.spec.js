import React from 'react'

import { renderWithTheme } from '@zap/test/unit/__helpers__/renderWithTheme'
import { QRCode } from 'components/UI'

describe('component.UI.QRCode', () => {
  it('should render correctly', () => {
    const tree = renderWithTheme(<QRCode value="qwerty" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
