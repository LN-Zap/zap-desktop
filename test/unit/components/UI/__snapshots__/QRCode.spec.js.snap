// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.QRCode should render correctly 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 150px;
  height: 150px;
}

.c0 {
  position: relative;
  width: 180px;
  height: 180px;
  display: inline-block;
}

.c2 {
  position: relative;
  width: 180px;
  height: 180px;
}

.c3 {
  display: inline-block;
  width: 180px;
  height: 180px;
  border: 1px solid #fd9800;
  border-radius: 10%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-clip-path: polygon(0 0,25% 0,25% 25%,0 25%);
  clip-path: polygon(0 0,25% 0,25% 25%,0 25%);
}

.c4 {
  display: inline-block;
  width: 180px;
  height: 180px;
  border: 1px solid #fd9800;
  border-radius: 10%;
  position: absolute;
  top: 0;
  right: 0;
  -webkit-clip-path: polygon(75% 0,100% 0%,100% 25%,75% 25%);
  clip-path: polygon(75% 0,100% 0%,100% 25%,75% 25%);
}

.c5 {
  display: inline-block;
  width: 180px;
  height: 180px;
  border: 1px solid #fd9800;
  border-radius: 10%;
  position: absolute;
  bottom: 0;
  left: 0;
  -webkit-clip-path: polygon(0 75%,25% 75%,25% 100%,0 100%);
  clip-path: polygon(0 75%,25% 75%,25% 100%,0 100%);
}

.c6 {
  display: inline-block;
  width: 180px;
  height: 180px;
  border: 1px solid #fd9800;
  border-radius: 10%;
  position: absolute;
  bottom: 0;
  right: 0;
  -webkit-clip-path: polygon(75% 75%,100% 75%,100% 100%,75% 100%);
  clip-path: polygon(75% 75%,100% 75%,100% 100%,75% 100%);
}

.c7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-style: solid;
  border-color: white;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-filter: none;
  filter: none;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: white;
  border-style: solid;
  border-color: white;
  position: absolute;
  top: calc(25% - 2px);
  left: calc(25% - 2px);
  -webkit-filter: blur(4px);
  filter: blur(4px);
  width: 50%;
  height: 50%;
  opacity: 0;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c8 {
  border-style: solid;
  border-color: white;
  border-width: 8px;
}

<div
  className="c0 c1"
  size={150}
>
  <div
    className="c2 c1"
    size={150}
  >
    <div
      className="c3 c1"
      size={150}
    />
    <div
      className="c4 c1"
      size={150}
    />
    <div
      className="c5 c1"
      size={150}
    />
    <div
      className="c6 c1"
      size={150}
    />
  </div>
  <div
    className="c7"
  >
    <svg
      className="c8"
      height={150}
      shapeRendering="crispEdges"
      viewBox="0 0 21 21"
      width={150}
    >
      <path
        d="M0,0 h21v21H0z"
        fill="white"
      />
      <path
        d="M0 0h7v1H0zM8 0h1v1H8zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM8 1h2v1H8zM11 1h1v1H11zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h5v1H8zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM10 3h1v1H10zM12 3h1v1H12zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM9 4h4v1H9zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM8 5h1v1H8zM10 5h1v1H10zM12 5h1v1H12zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM0 8h2v1H0zM4 8h3v1H4zM12 8h1v1H12zM15 8h1v1H15zM17,8 h4v1H17zM1 9h4v1H1zM8 9h1v1H8zM12 9h3v1H12zM17 9h2v1H17zM0 10h1v1H0zM4 10h3v1H4zM10 10h1v1H10zM12 10h2v1H12zM18 10h2v1H18zM1 11h1v1H1zM3 11h2v1H3zM7 11h6v1H7zM15 11h1v1H15zM19 11h1v1H19zM1 12h1v1H1zM3 12h5v1H3zM10 12h2v1H10zM14 12h3v1H14zM19,12 h2v1H19zM8 13h2v1H8zM11 13h2v1H11zM17 13h2v1H17zM0 14h7v1H0zM11 14h1v1H11zM14 14h3v1H14zM19 14h1v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h2v1H8zM13 15h2v1H13zM16 15h1v1H16zM20,15 h1v1H20zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM8 16h1v1H8zM10 16h1v1H10zM12 16h3v1H12zM16 16h2v1H16zM20,16 h1v1H20zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h2v1H9zM12 17h5v1H12zM18,17 h3v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM12 18h2v1H12zM16 18h3v1H16zM0 19h1v1H0zM6 19h1v1H6zM8 19h5v1H8zM15 19h2v1H15zM0 20h7v1H0zM8 20h2v1H8zM11 20h1v1H11zM14 20h2v1H14zM17 20h1v1H17zM20,20 h1v1H20z"
        fill="black"
      />
    </svg>
  </div>
  <div
    className="c9"
  />
</div>
`;
